docs/diagrams/src/architecture.puml

```diff
@@ -0,0 +1,98 @@
+@startuml app_architecture
```

Comment:
I guess this is okay for now, If I find the time I will try to make a revised version of this